src/Resources/ResourceManager/Components/ApiVersionHelper.cs

```diff
@@ -127,11 +128,12 @@ private static string[] GetApiVersionsForResourceType(IAzureContext context, str
                     cancellationToken: cancellationToken),
                 cancellationToken: cancellationToken);
 
+            string topLevelResourceType = ResourceTypeUtility.GetTopLevelResourceType(resourceType);
```

Comment:
Don't we want to do this only in the case where the resource type is not found initially (that is, if the original query returns nothing?